### DISCUSSION TITLE
Warning message linux/arm/v7

### DISCUSSION THREAD

**Discussion by IssueFindings** [dani-garcia/vaultwarden#1264] [General]
Hello,

I run Bitwarden_rs through docker. Each time I launch it, I can read this warning message `WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm/v7) and no specific platform was requested` What that really means ? How I can fix it ?

Have a nice day.

**Comment by BlackDex** [comment]
You have the give a bit more information then this. Docker normally checks for the platform it self. It seems that you force a specific platform.

So what we need to see if there is something obvious is what docker create/run command you use, what platform/hardware and which os.

**Comment by IssueFindings** [comment]
Hello,
Sorry for the lack of informations. So, I use `Docker version 20.10.0, build 7287ab3` on `Linux raspberrypi 5.4.72-v7l+ #1356 SMP (...)  armv7l GNU/Linux`. I use the common way to install Docker.
Have a nice day

**Comment by BlackDex** [comment]
You somehow have the amd64 image selected instead of the armv7 one, i know this can sometimes happen and i don't know why, since that is a docker issue not bitwarden_rs.
You could try to add `--platform linux/arm/v7` but i'm not sure if that will help.

**Reply by AimanAliyas** [reply]
Hi, i'm running into a similar problem and can you tell me where exactly do i need to add tha --platform linux/arm/v7 , is it in the Dockerfile or in the terminal?

**Reply by adv0r** [reply]
same

**Comment by TheGreatestJannet** [comment]
I have the same issue. It seems to be related to the latest docker update

**Comment by TheGreatestJannet** [comment]
Running: `docker run -e SIGNUPS_ALLOWED=false -e ADMIN_TOKEN=password -e WEBSOCKET_ENABLED=true -p 8081:80 -p 8082:3012 -v /opt/bw-data:/data/ --rm --name bitwarden bitwardenrs/server:latest --platform linux/arm/v7` 
I get: `Command not found: --platform`

**Reply by westechworkshop** [reply]
put the "--platform linux/arm/v7" right after "docker run " and before "-e SIGNUPS....."

**Reply by 3bars** [reply]
this worked, thx you

**Reply by sgerodes** [reply]
worked for me too. Thank you!

**Comment by jjlin** [comment]
`--platform linux/arm/v7` needs to go before `bitwardenrs/server:latest`.

Do you get the same issue if you run another image? For example:

    docker run -it --rm alpine

**Comment by TheGreatestJannet** [comment]
Running that I get: `docker: Error response from daemon: image with reference bitwardenrs/server:latest was found but does not match the specified platform: wanted linux/arm/v7, actual: linux/amd64`

**Reply by LogicKahanHai** [reply]
Did you get the solution for this? I am facing the same issue.
Thanks

**Reply by BlackDex** [reply]
Try to update your docker tools.
https://docs.docker.com/engine/install/

**Comment by TheGreatestJannet** [comment]
This also prevents the systemd job from working and requires bitwardenrs to be manually started every time you reboot.

**Comment by jjlin** [comment]
This is almost certainly a regression in Docker 20.10 then, as `alpine` is an official Docker image. I'd recommend rolling back to Docker 19.03.x until this is fixed.

**Comment by TheGreatestJannet** [comment]
Is there anywhere to report the bug to the developers? Looking on the forums it seems that no one has mentioned anything about it so I'm not even sure if they know about it.

**Comment by jjlin** [comment]
https://github.com/moby/moby is probably the right place. Also, did you actually try running the `alpine` image? Just noticed your error message wasn't for `alpine`.

**Comment by TheGreatestJannet** [comment]
No I don't get an error message for alpine.

**Comment by EchoDev** [comment]
I can confirm I have the same issue

latest mysql build with armv8

**Comment by TheGreatestJannet** [comment]
This issue has been fixed in the latest update of docker.

**Reply by IssueFindings** [reply]
Hello,
Could you show me how you fix that ? I found nothing about that.
Have a nice day.

**Reply by nilquera** [reply]
I installed docker with the latest [convenience script](https://docs.docker.com/engine/install/debian/#install-using-the-convenience-script) and the problem persists.

**Comment by jjlin** [comment]
I did some testing on my RPi 2, and this is an artifact of the build process. We cross-build on `amd64` for the other arches, and since Docker doesn't seem to expose a way to directly specify the arch of the resulting image, it ends up being set to the build arch (`amd64`) even though it's not.

This is just a warning message, and the image still works as it should, but I'm working on fixing the build process so the warning won't come up.

**Reply by DrDeath** [reply]
Just testet with the new bitwardenrs/server:testing  on a raspberry 3 
```
Linux raspberrypi 5.4.83-v7+ #1379 SMP Mon Dec 14 13:08:57 GMT 2020 armv7l GNU/Linux
docker version
Client: Docker Engine - Community
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:18:13 2020
 OS/Arch:           linux/arm
 Context:           default
 Experimental:      true
```

https://github.com/dani-garcia/bitwarden_rs/commit/37cc0c34cf5732cbe7b0d4aea622acdd615e3ef6


Runs now without the mentioned warning message.

**Reply by Cyrilnorbr365** [reply]
It worked with the parameter --platform, thanks for your help.

`docker run --platform linux/arm64 -dp 3001:3000 image_name`

**Comment by owolabiezekiel** [comment]
I have this same issue with my M1 mac. I get this error """The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested"""

Please what can I do?

**Reply by westechworkshop** [reply]
put "--platform linux/arm/v7" right after "docker run " and see if that works. If you already have that somewhere else in your command, move it to the place I just said because those types of arguments go before the actual command.

**Reply by owolabiezekiel** [reply]
It still isn't working. I am tired right now. really tired

**Reply by westechworkshop** [reply]
I don’t know what to tell you then. I also had the same issue doing this, but I somehow managed to make it work. You know, it’s one of those things where you don’t know how you did it but you did it. The code I ran is somewhere lost in the digital universe.

**Reply by owolabiezekiel** [reply]
Yeah, you have been super helpful. If I ever get a solution, I will make sure to post it here. Thanks

**Reply by westechworkshop** [reply]
Glad I could help inspire ideas.

**Reply by syamkarthik24** [reply]
iam tired of this i have been trying since 2 days and it is still not working. Did it work for you?

**Reply by BlackDex** [reply]
@syamkarthik24 it could help of you tell us what you have been trying, and what the results are, which hardware etc..

**Reply by jelade** [reply]
I am having the same issue with my M1 2023

**Comment by tjayadam** [comment]
I've also been receiving this error. For me, I have been trying to install QEMU emulator to emulate an ARM architecture. I can't install QEMU using `$ sudo install qemu qemu-user-static qemu-user binfmt-support` and so when I try to run docker I keep getting the warning 'The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested'. Any ideas on how to run an emulator or resolve the warning error for this instance?

**Reply by mydevice** [reply]
root@rancher:~# docker run -d --restart=unless-stopped \
>     -p 80:80 -p 443:443 \
>     --privileged \
>     rancher/rancher:v2.5.8-linux-arm64
Unable to find image 'rancher/rancher:v2.5.8-linux-arm64' locally
v2.5.8-linux-arm64: Pulling from rancher/rancher
673aeee5c81c: Pull complete
018b2790219d: Pull complete
509c77ce92ad: Pull complete
f1c29a69111b: Pull complete
b853a1c0402c: Pull complete
abbb9fe0e459: Pull complete
986cd8c62b49: Pull complete
99e085554b5f: Pull complete
3633dd7934b7: Pull complete
c431bff57b35: Pull complete
f2958273eb21: Pull complete
3b1e8c21c1ec: Pull complete
06ef878a8b07: Pull complete
592a14ff376a: Pull complete
4fea48bc0509: Pull complete
71112829f26c: Pull complete
e4248eeb8379: Pull complete
b0ba0feb5991: Pull complete
164090451cd1: Pull complete
Digest: sha256:d8802d9fd23eb062ab0a30c7e53f7c995b2e4df981299b6c14d21347f30ceb86
Status: Downloaded newer image for rancher/rancher:v2.5.8-linux-arm64
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
9fb65fda852a116f8e14e7bc3df7c938a0c09d5385f42726a333f32da7ad0618

**Comment by onahte** [comment]
It seems building the image on an M1 results in OS architecture of linux/arm64/v8. The OS architecture of the build and the run must be compatible.
Try: docker buildx build --platform linux/amd64 -t <image_name>

**Reply by kiranojhanp** [reply]
Thank you
